### PR TITLE
applicationInstallationController: set conditions

### LINKED ISSUE #10546: ApplicationInstallationController should set the conditions
### What happened?
the controller does not set the condition so the UI can not get information about the installation status.

### Expected behavior

the condition for RetrivedManfifest and Ready should be set by the controller

### PR DESCRIPTION
**Pull request by vgramer**
**What does this PR do / Why do we need it**:
this pr set the conditions in the status. the following change have been made:

* conditions type change from slice to `map[appskubermaticv1.ApplicationInstallationConditionType]appskubermaticv1.ApplicationInstallationCondition` for convenience when manipulating conditions.
* ApplicationInstallationCondition now contains the `LastTransitionTime` and `LastHeartbeatTime` (useful when an application is upgraded)
*  The `Apply`method of `Installer` was previously downloading the source and installing/upgrading the application. This method has been split into 2 functions. `DownloadSource` and  `Apply`  in order to be able to set the condition `ManifestRetrived`

On API side:
* add `method` field  in `apiv2.ApplicationInstallationStatus`.  it was previously in applicationVersion but when mode it to spec level, we forget to update the API part.
* when converting type from external to internal, we don't set the status as it makes no sense because it is only set by the controller and ignored for creating / updating operations.  Moreover, it avoids conversion problem when Do POST (see #10502). 

**Does this PR close any issues?**:
Fixes  #10546 #10408

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/apps.kubermatic/v1/application_installation.go [hunk 1]
```diff
@@ -125,8 +125,9 @@ type Version struct {
 // ApplicationInstallationStatus denotes status information about an ApplicationInstallation.
 type ApplicationInstallationStatus struct {
 	LastUpdated metav1.Time `json:"lastUpdated,omitempty"`
```
Comment (vgramer) [inline pkg/apis/apps.kubermatic/v1/application_installation.go]: @SimonTheLeg i think we can remove this field as condition as `LastTransitionTime` and  `LastHeartbeatTime`. field. WDYT?

File: pkg/handler/v2/application_installation/application_installation.go [hunk 2]
```diff
@@ -184,11 +204,6 @@ func convertExternalToInternal(app *apiv2.ApplicationInstallation) *appskubermat
 			Namespace: app.Namespace,
 		},
 		Spec: *app.Spec,
-		Status: appskubermaticv1.ApplicationInstallationStatus{
```
Comment (vgramer) [inline pkg/handler/v2/application_installation/application_installation.go:187]: status is only set by the controller and will be ignored by create/update operation. So it makes no sense to set it.

IMHO we should remove the status from `createApplicationInstallationReq` / `updateApplicationInstallationReq`

@SimonTheLeg WDYT

Comment (vgramer) [reply pkg/handler/v2/application_installation/application_installation.go:187]: there already is an issue for that #10365

File: pkg/apis/apps.kubermatic/v1/application_installation.go (hunk 1 above)
Comment (vgramer) [reply pkg/apis/apps.kubermatic/v1/application_installation.go]: as discussed in meeting we can remove this field

File: pkg/controller/user-cluster-controller-manager/application-installation-controller/controller.go [hunk 3]
```diff
@@ -219,23 +220,62 @@ func (r *reconciler) getApplicationVersion(appInstallation *appskubermaticv1.App
...
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	oldAppInstallation = appInstallation.DeepCopy()
+	statusUpdater, installErr := r.appInstaller.Apply(ctx, log, r.seedClient, r.userClient, appInstallation, appSourcePath)
+
+	if installErr != nil {
+		r.setCondition(appInstallation, appskubermaticv1.Ready, corev1.ConditionFalse, "InstallationFailed", installErr.Error())
+	} else {
+		r.setCondition(appInstallation, appskubermaticv1.Ready, corev1.ConditionTrue, "InstallationSuccessful", "application sudccessfuly install or upgraded")
```
Comment (lsviben) [inline pkg/controller/user-cluster-controller-manager/application-installation-controller/controller.go]: typo
```suggestion
		r.setCondition(appInstallation, appskubermaticv1.Ready, corev1.ConditionTrue, "InstallationSuccessful", "application successfully installed or upgraded")
```

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**